src/Pact/Utils.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Pact.Utils
+( decodeB64UrlNoPaddingText
+, encodeB64UrlNoPaddingText
+)where
+
+
+import qualified Data.ByteString as B
+import qualified Data.ByteString.Base64.URL as B64U
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+
+
+-- | Decode a binary value from a textual base64-url without padding
+-- representation. A 'Base64DecodeException' is thrown if the input is not a
+-- valid base64-url without padding encoding.
+--
+decodeB64UrlNoPaddingText :: T.Text -> Either String B.ByteString
```

Comment:
This should be in `Pact.Types.Util` and harmonized with base64 functions therein